### PR TITLE
297 asset patterns regex

### PR DESCRIPTION
**Pull request by 80vs90**

### CODE REVIEW

File: gips/data/core.py [hunk 1]
```diff
@@ -309,7 +309,11 @@ def discover(cls, tile, date, asset=None):
             assets = cls._assets.keys()
         found = []
         for a in assets:
-            files = glob.glob(os.path.join(tpath, cls._assets[a]['pattern']))
+            if os.path.isdir(tpath):
+                pattern_re = re.compile(cls._assets[a]['pattern'])
+                files = [os.path.join(tpath, f) for f in os.listdir(tpath) if os.path.isfile(os.path.join(tpath, f)) and pattern_re.match(f)]
```
Comment (ircwaves) [inline gips/data/core.py]: To keep it PEP8-ish, lines should be less than 80 characters

File: gips/data/merra/merra.py [hunk 2]
```diff
@@ -103,7 +102,7 @@ class merraAsset(Asset):
             'shortname': 'M2T1NXSLV',
             'description': '2d,1-Hourly,Time-Averaged,Single-Level,Assimilation,Single-Level Diagnostics V5.12.4',
             'url': 'https://goldsmr4.gesdisc.eosdis.nasa.gov/data/MERRA2/M2T1NXSLV.5.12.4',
-            'pattern': _asset_pattern.format(name='tavg1_2d_slv_Nx'),
+            'pattern': r'MERRA2_\d\d\d\.tavg1_2d_slv_Nx\.\d{4}\d{2}\d{2}\.nc4',
```
Comment (ircwaves) [inline gips/data/merra/merra.py]: Seems like this should continue the `_asset_re_pattern.format(name='tavg1_2d_slv_Nx')` style, for DRYness.

File: gips/data/merra/merra.py [hunk 3]
```diff
@@ -89,7 +89,8 @@ class merraAsset(Asset):
     }
 
     _bandnames = ['%02d30GMT' % i for i in range(24)]
-    _asset_re_pattern = "MERRA2_\d\d\d\.{name}\.%04d%02d%02d\.nc4"
+    _asset_re_pattern = 'MERRA2_\d\d\d\.%s\.\d{4}\d{2}\d{2}\.nc4'
```
Comment (80vs90) [inline gips/data/merra/merra.py]: I know that the `format()` method is preferred, but that plays havoc with the regex repeat syntax.

Comment (ircwaves) [inline gips/data/merra/merra.py]:
```
except TwentyTwo:
    pass
```
Thinking about it...

Comment (bhbraswell) [inline gips/data/merra/merra.py]: say what?

Comment (ircwaves) [inline gips/data/merra/merra.py]: Translates in my head roughly to:  _it's a catch 22, so we carry on_.

**Comment by ircwaves** [comment]
Close #297

**Pull request merged**